### PR TITLE
Update optional-test-infra doc

### PR DESCRIPTION
**Pull request by PatrickXYS**
**Which issue is resolved by this Pull Request:**
Resolves #

**Description of your changes:**
1. Update optional-test-infra documentation
2. Make root README clean up, and divide documentation into oss-test-infra and optional-test-infra parts.

**Checklist:**

If PR related to **Optional-Test-Infra**,
- [ ] Changes have been generated to `aws/GitOps` folder: 
    1. `cd aws`
    2. `make optional-generate`
    3. `make optional-test`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,1110 +1,8 @@
-<!-- START doctoc generated TOC please keep comment here to allow auto update -->
```
Comment (PatrickXYS) [inline README.md:1]: @Bobgy I cleaned up the README and divided this into two parts, oss-test-infra and optional-test-infra.

Let me know if you want us to better organize the documentation

File: images/Dockerfile.py3.aws [hunk 2]
```diff
@@ -14,9 +14,6 @@ RUN curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py && python get-pip.py
 RUN python3.8 -m pip install \
     filelock \
     fire \
-    google-api-python-client \
```
Comment (PatrickXYS) [inline images/Dockerfile.py3.aws:17]: I think this is unnecessary to have those dependencies in optional-test-infra worker image.

File: README.md (hunk 1 above)
Comment (Jeffwan) [reply README.md:1]: project root level does need a README.md. I would suggest to move some content out of the root README.md instead of deleting it

Comment (PatrickXYS) [reply README.md:1]: I didn't delete those doc but moved it into gcp_README.md. 

Maybe I can add some common content back to README.md, what do you think?

Comment (Jeffwan) [reply README.md:1]: sounds good. Seems some contents are general information here.

Comment (PatrickXYS) [reply README.md]: Moved contents back to root README

**Comment by PatrickXYS** [comment]
/cc @Bobgy @Jeffwan 
Can you take another look?

**Comment by PatrickXYS** [comment]
I'll merge it as of now, if we have any other concerns, we can open another PR to update doc

**Pull request merged**